frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditOptionList/OptionListEditor.test.tsx

```diff
@@ -0,0 +1,165 @@
+    await renderOptionListEditorAndWaitForSpinnerToBeRemoved();
+
+    const btnOpen = screen.getByRole('button', {
+      name: textMock('ux_editor.modal_properties_code_list_open_editor'),
+    });
+
+    expect(btnOpen).toBeInTheDocument();
+  });
+
+  it('should open Dialog', async () => {
+    const user = userEvent.setup();
+    await renderOptionListEditorAndWaitForSpinnerToBeRemoved();
+
+    await openModal(user);
+
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+  });
+
+  it('should close Dialog', async () => {
+    const user = userEvent.setup();
+    await renderOptionListEditorAndWaitForSpinnerToBeRemoved();
+
+    await openModal(user);
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should call doReloadPreview when editing', async () => {
+    const user = userEvent.setup();
+    const doReloadPreview = jest.fn();
+    await renderOptionListEditorAndWaitForSpinnerToBeRemoved({
+      previewContextProps: { doReloadPreview },
+    });
+
+    await openModal(user);
+    const textBox = screen.getByRole('textbox', {
+      name: textMock('ux_editor.modal_properties_code_list_item_description', { number: 2 }),
+    });
+    await user.type(textBox, 'test');
+
+    await waitFor(() => expect(doReloadPreview).toHaveBeenCalledTimes(1));
+  });
+
+  it('should call updateOptionList with correct parameters when closing Dialog', async () => {
+    const user = userEvent.setup();
+    await renderOptionListEditorAndWaitForSpinnerToBeRemoved();
+    const expectedResultAfterEdit: Option[] = [
+      { value: 'test', label: 'label text', description: 'description', helpText: 'help text' },
+      { value: 2, label: 'label number', description: 'test', helpText: null },
+      { value: true, label: 'label boolean', description: null, helpText: null },
+    ];
+
+    await openModal(user);
+    const textBox = screen.getByRole('textbox', {
+      name: textMock('ux_editor.modal_properties_code_list_item_description', { number: 2 }),
+    });
+    await user.type(textBox, 'test');
+
+    await waitFor(() => expect(queriesMock.updateOptionList).toHaveBeenCalledTimes(1));
+    expect(queriesMock.updateOptionList).toHaveBeenCalledWith(
+      org,
+      app,
+      mockComponent.optionsId,
+      expectedResultAfterEdit,
+    );
+  });
+});
+
+const openModal = async (user: UserEvent) => {
+  const btnOpen = screen.getByRole('button', {
+    name: textMock('ux_editor.modal_properties_code_list_open_editor'),
+  });
+  await user.click(btnOpen);
+};
+
+const renderOptionListEditor = ({ previewContextProps = {}, queries = {} } = {}) => {
+  return renderWithProviders(
+    <OptionListEditor
+      component={{
+        ...mockComponent,
+      }}
```

Comment:
Since we are not modifying the component in any of the tests we can pass the mockComponent directly. Or pass only the mockOptionListId if we adjust the component 😬